buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -346,6 +409,7 @@ private void configure()  {
             config.put("cluster.initial_master_nodes", "[" + nodeName + "]");
         }
         try {
+            Files.delete(configFile);
```

Comment:
I've seen a number of instances of us creating big strings in memory. `LoggedExec` is another, more problematic instance of this. Individually, these are not big issues but we need to be careful about this stuff as the build becomes more complex, and we rely more on the daemon, memory pressure will become an issue. We should try and keep this in mind and prefer streaming this stuff when possible.

Reply:
LoggedExec is a special case. We could redirect output to files, but most of the exec calls are tiny invocations, so the I/O overhead to disk is much more than writing into the heap. But I agree here we should consider writing the file through a stream instead of building up in a String.

Reply:
Agreed. If it hasn't been problematic yet it may just not be an issue. We still have a situation where a bad execution could potentially create a load of output that blows out the heap. We could pretty simply put some safeguards here to ensure the output doesn't exceed some configured threshold and truncate if it does. Separate discussion 😄